### ISSUE TITLE
l10n dev site is broken

### ISSUE THREAD

**Issue by flodolo** [mozilla/testpilot#3596]
https://testpilot-l10n.dev.mozaws.net/

Only loads a background image

```
<unavailable> vendor.19577a63.js:1:382546
TypeError: p is undefined app.cb5731a3.js:1:33254 
```

**Comment by flodolo** [comment]
Side note: any chance to improve the l10n workflow? Since we moved to a l10n branch, things have gotten worse.

For example: today I believe we're going to branch out the next prod release, with new experiments, and we only exposed those strings 2 days ago (and now the dev server for testing is broken). We had a case with strings untranslated on prod for weeks, because master and l10n weren't merged.

**Comment by flodolo** [comment]
@johngruen and @clouserw Thoughts? 

I confess the choice of using a l10n branch might have simplified things for development, but not improved the situation for l10n. And I've just got the first complaint (justified), asking why the website didn't have their translation live, given that they were completed several days ago

**Comment by ghost** [comment]
Hi Flod.  We're working on putting Test Pilot in "auto pilot" mode, meaning, we're going to stop adding features and tweaking the site, and instead focus on experiments themselves.  We're close, you can see what's left at https://github.com/mozilla/testpilot/projects/12 .  Our development flow hasn't been the normal rhythm because of this initiative and people rotating off to work on experiments.  Once we finish the auto pilot push, I expect work on Test Pilot to stop, aside from new experiment content, which we should merge to l10n immediately after it lands.

Last week we broke our dev server with a syntax error (which broke the l10n server as well) and then I never pushed the fix to the l10n server.  We only do triage once a week so that was unfortunate timing on this issue being filed.  Please ping any of us on Slack or IRC to get more notice on a critical bug like this.

Sorry for the delay and mistakes here.

I've merged master to l10n which brought the l10n site back to life.

In a post-autopilot world, we'll need to continue to merge between master and l10n and push to production, but that's a simple operation and well documented.  I think we could even do it every week since it doesn't need to center around a sprint.  I'll make some calendar reminders to make sure that happens on a regular schedule, and I think that consistency will address this.

I'm going to close this because the server is back, but we can keep discussing.

**Comment by flodolo** [comment]
Thanks for fixing the issue. I'll keep an eye on this in the coming weeks.

A good idea would be to make sure master->l10n updates are merged as soon as possible when new strings land.

**Comment by lmorchard** [comment]
> A good idea would be to make sure master->l10n updates are merged as soon as possible when new strings land.

Well, the problem we were trying to solve is that new strings "land" in master. But then maybe we decide we need to further iterate and those new strings change several times. So they end up being unstable for a bit. Sometimes changing several times in a day in a few cases.

We were trying to develop a habit with the separate l10n branch to only merge once strings stabilized. But, as habits aren't automated, we forget to do the merge sometimes or merge too close to a production release.

I wonder if rather than manually merging master to l10n if we instead had a bot that merged or force-updated l10n every week or 2 weeks? Then, the strings in l10n would be defacto stable, if only because we enforce a fixed automated window to merge strings and present them to localizers.

**Comment by flodolo** [comment]
There's also a schedule/planning concern: strings for the new experiments landed on Friday (May 25) in #3584, with a production cut-off on Thursday (May 31). In this case, we lost several days because of the missed merge to l10n (it happened on May 29). A weekly merge would not likely work in this case, and we still need the merge back to master.

The worst part is that on Tuesday there's the big communication outreach, and we're out with unlocalized text on the website, and that's not going to be fixed for the following 2 weeks. Talking with Will, it looks like the release cycle might get faster in the future, which, at least, could mitigate the issue.

**Comment by lmorchard** [comment]
> There's also a schedule/planning concern

Yeah, I don't think there's any automation that can help us there. We just really have to land strings early when we have a big marketing push coming :(
